src/apps/api/views/submissions.py

```diff
@@ -96,6 +99,37 @@ def has_admin_permission(self, user, submission):
         competition = submission.phase.competition
         return user.is_superuser or user in competition.all_organizers
 
+    @action(detail=True, methods=('POST', 'DELETE'))
+    def submission_leaderboard_connection(self, request, pk):
+        submission = self.get_object()
+        phase = submission.phase
+
+        if not (request.user.is_superuser or request.user == submission.owner or phase.competition.collaborators.filter(pk=request.user.pk).exists()):
```

Comment:
```suggestion
        is_collaborator = phase.competition.collaborators.filter(pk=request.user.pk).exists())
        if not (request.user.is_superuser or request.user == submission.owner or is_collaborator:
```

Let's shorten this guy just a bit by breaking it up into two lines. I'm absolutely imposing my personal opinions on you at this point.